### PR TITLE
Documentation: Reorganize some section

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
1.note, socketcan and syslog should put into driver section
2.nsh should put into application section

### CODE REVIEW

**Review by btashton — APPROVED** [review]

File: Documentation/components/drivers/special/index.rst [hunk 1]
```diff
@@ -22,9 +22,11 @@ following section.
 .. toctree::
   :caption: Supported Drivers
 
+  syslog.rst
   spi.rst
   i2c.rst
   ethernet.rst
+  socketcan.rst
```
Comment (xiaoxiang781216) [inline Documentation/components/drivers/special/index.rst:29]: @btashton and @v01d do you know why syslog.rst, socketcan.rst can't show on the website?
https://nuttx.apache.org/docs/latest/components/drivers/special/index.html
even the source show that they are there:
https://nuttx.apache.org/docs/latest/_sources/components/drivers/special/index.rst.txt
nsh has the similar issue, but note.rst don't, very strange.

Comment (btashton) [reply Documentation/components/drivers/special/index.rst:29]: https://nuttx.apache.org/docs/latest/components/drivers/special/socketcan.html

I see it here?

Comment (xiaoxiang781216) [reply Documentation/components/drivers/special/index.rst:29]: My sceenshot at https://nuttx.apache.org/docs/latest/components/drivers/special/index.html:
image
My sceenshot at https://nuttx.apache.org/docs/latest/components/drivers/special/socketcan.html:
image
Very strange.

Comment (v01d) [reply Documentation/components/drivers/special/index.rst:29]: @xiaoxiang781216 try to do "CTRL+SHIT+R", this forces the browser to reload and not use cache. I've seen this happen some times on Firefox at least. I wonder if we need some extra tag on the HTMLs to make the server know about the changed HTML

Comment (xiaoxiang781216) [reply Documentation/components/drivers/special/index.rst:29]: Yes, reload can work! Thanks, @v01d.

**Pull request merged**